@xen-orchestra/backups/_VmBackup.js

```diff
@@ -321,14 +322,17 @@ exports.VmBackup = class VmBackup {
     this._fullVdisRequired = fullVdisRequired
   }
 
-  async run() {
+  async run($defer) {
```

Comment:
To keep this implementation more intuitive, I suggest:

```js
run = defer(this.run)
asyn run($defer) {
```

Reply:
You're right.

I did not do that because it wraps the method for each instance instead of a only once for the class.
But there is no good way to do this apart from decorators, so I guess your approach is good enough for now. :slightly_smiling_face: